lib/node_modules/@stdlib/lapack/base/dgttrf/lib/base.js

```diff
@@ -0,0 +1,180 @@
+* @returns {integer} status code
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+* var Int32Array = require( '@stdlib/array/int32' )
+*
+* var DL = new Float64Array( [ 1.0, 1.0 ] );
+* var D = new Float64Array( [ 2.0, 3.0, 1.0 ] );
+* var DU = new Float64Array( [ 1.0, 1.0 ] );
+* var DU2 = new Float64Array( [ 0.0 ] );
+* var IPIV = new Int32Array( [ 0, 0, 0 ] );
+*
+* dgttrf( 3, DL, 1, 0, D, 1, 0, DU, 1, 0, DU2, 1, 0, IPIV, 1, 0 );
+* // DL => <Float64Array>[ 0.5, 0.4 ]
+* // D => <Float64Array>[ 2, 2.5, 0.6 ]
+* // DU => <Float64Array>[ 1, 1 ]
+* // DU2 => <Float64Array>[ 0 ]
+* // IPIV => <Int32Array>[ 0, 1, 2 ]
+*/
+function dgttrf( N, DL, strideDL, offsetDL, D, strideD, offsetD, DU, strideDU, offsetDU, DU2, strideDU2, offsetDU2, IPIV, strideIPIV, offsetIPIV ) { // eslint-disable-line max-len, max-params
+	var fact;
+	var temp;
+	var idu2;
+	var idu;
+	var idl;
+	var id;
+	var ip;
+	var i;
+
+	// Quick return if possible
+	if ( N === 0 ) {
+		return 0;
+	}
+
+	idl = offsetDL;
+	id = offsetD;
+	idu = offsetDU;
+	idu2 = offsetDU2;
+	ip = offsetIPIV;
+
+	// Initialize ith element of IPIV as i
+	for ( i = 0; i < N; i++ ) {
+		IPIV[ ip ] = i;
+
+		if ( i < N-2 ) {
+			// Initialize ith element of DU2 as 0
+			DU2[ idu2 ] = 0;
+		}
+
+		ip += strideIPIV;
+		idu2 += strideDU2;
+	}
+
+	// Set the pointers to the starting indices
+	idu2 = offsetDU2;
+	ip = offsetIPIV;
+
+	for ( i = 0; i < N-2; i++ ) {
+		if ( abs( D[ id ] ) >= abs( DL[ idl ] ) ) { // No row interchange required, eleminate ith element of DL
+			if ( D[ id ] !== 0.0 ) {
+				fact = DL[ idl ] / D[ id ];
+				DL[ idl ] = fact;
+				D[ id + strideD ] = D[ id + strideD ] - ( fact*DU[ idu ] );
+			}
+		} else { // Interchange the ith and (i+1)th rows and eliminate ith element of DL
+			fact = D[ id ] / DL[ idl ];
+			D[ id ] = DL[ idl ];
+			DL[ idl ] = fact;
+			temp = DU[ idu ];
+			DU[ idu ] = D[ id + strideD ];
+			D[ id + strideD ] = temp - ( fact*D[ id + strideD ] );
+			DU2[ idu2 ] = DU[ idu + strideDU ];
+			DU[ idu + strideDU ] = -fact*DU[ idu + strideDU ];
+			IPIV[ ip ] = i + 1;
+		}
+
+		id += strideD;
+		idl += strideDL;
+		idu += strideDU;
+		idu2 += strideDU2;
+		ip += strideIPIV;
+	}
+
+	if ( N > 1 ) {
+		i = N - 2;
+		idl = offsetDL + ( strideDL * i );
+		id = offsetD + ( strideD * i );
+		idu = offsetDU + ( strideDU * i );
+		idu2 = offsetDU2 + ( strideDU2 * i );
+		ip = offsetIPIV + ( strideIPIV * i );
+
+		if ( abs( D[ id ] ) >= abs( DL[ idl ] ) ) {
+			if ( D[ id ] !== 0 ) {
+				fact = DL[ idl ] / D[ id ];
+				DL[ idl ] = fact;
+				D[ id + strideD ] = D[ id + strideD ] - ( fact * DU[ idu ] );
+			}
+		} else {
+			fact = D[ id ] / DL[ idl ];
+			D[ id ] = DL[ idl ];
+			DL[ idl ] = fact;
+			temp = DU[ idu ];
+			DU[ idu ] = D[ id + strideD ];
+			D[ id + strideD ] = temp - ( fact*D[ id + strideD ] );
+			IPIV[ ip ] = i + 1;
+		}
+	}
+
+	id = offsetD;
+
+	// Check for a 0 on the diagonal of U
```

Comment:
```suggestion
	// Check for a 0 on the diagonal of U
```

I suggest updating this comment to describe what you are trying to determine mathematically.